src/app/app-platform/ContentAppPlatform.cpp

```diff
@@ -388,14 +389,131 @@ uint32_t ContentAppPlatform::GetPincodeFromContentApp(uint16_t vendorId, uint16_
+            entry.AddTarget(nullptr, target);
+        }
+
+        bindings.push_back(Binding::Structs::TargetStruct::Type{
+            .fabricIndex = kUndefinedFabricIndex,
+            .node        = MakeOptional(localNodeId),
+            .group       = NullOptional,
+            .endpoint    = MakeOptional(kLocalVideoPlayerEndpointId),
+            .cluster     = MakeOptional(kNoClusterIdSpecified),
+        });
+    }
+
+    ChipLogProgress(Controller, "Create speaker endpoint ACL and binding");
+    {
+        Access::AccessControl::Entry::Target target = { .flags    = Access::AccessControl::Entry::Target::kEndpoint,
+                                                        .endpoint = kLocalSpeakerEndpointId };
+        entry.AddTarget(nullptr, target);
+
+        bindings.push_back(Binding::Structs::TargetStruct::Type{
+            .fabricIndex = kUndefinedFabricIndex,
+            .node        = MakeOptional(localNodeId),
+            .group       = NullOptional,
+            .endpoint    = MakeOptional(kLocalSpeakerEndpointId),
+            .cluster     = MakeOptional(kNoClusterIdSpecified),
+        });
+    }
+
+    ChipLogProgress(Controller, "Create content app endpoints ACL and binding");
+    {
+        uint8_t index = 0;
+        while (index < CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT)
+        {
+            ContentApp * app = mContentApps[index++];
+            if (app == nullptr)
+            {
+                continue;
+            }
+
+            for (const auto & allowedVendor : app->GetApplicationBasicDelegate()->GetAllowedVendorList())
+            {
+                if (allowedVendor == targetVendorId)
+                {
+                    Access::AccessControl::Entry::Target target = { .flags    = Access::AccessControl::Entry::Target::kEndpoint,
+                                                                    .endpoint = app->GetEndpointId() };
+                    entry.AddTarget(nullptr, target);
+
+                    bindings.push_back(Binding::Structs::TargetStruct::Type{
+                        .fabricIndex = kUndefinedFabricIndex,
+                        .node        = MakeOptional(localNodeId),
+                        .group       = NullOptional,
+                        .endpoint    = MakeOptional(app->GetEndpointId()),
+                        .cluster     = MakeOptional(kNoClusterIdSpecified),
+                    });
+                }
+            }
+        }
+    }
+
+    ReturnErrorOnFailure(GetAccessControl().CreateEntry(nullptr, entry, nullptr));
+
+    for (const auto & binding : bindings)
+    {
+        Binding::Structs::TargetStruct::Type entries[1];
+        entries[0] = binding;
+        BindingListTypeInfo bindingList(entries);
+
+        ChipLogProgress(Controller, "Attempting to create Binding");
+        ReturnErrorOnFailure(
+            CreateBindingWithCallback(targetDeviceProxy, kTargetBindingClusterEndpointId, bindingList, successCb, failureCb));
```

Comment:
This part does not make sense to me.  We take a list.  Then for each entry in the list we create a length-1 list and write that to the binding attribute.  The outcome will be a binding list that just has the last entry from our initial list, no?  I doubt that's the goal here.

Was this code tested at all?

I suspect making `bindings` an `std::vector` and then creating a BindingListTypeInfo here from its `data()` and `size()` would work much better.

Reply:
Sorry, this is a draft PR. Added that to the title.

Reply:
Updated to perform a single attribute write with the entire list